### PR TITLE
gadget/device: introduce package which provides helpers for locations of things

### PR DESCRIPTION
**Pull request by bboozzoo**
Various helpers for getting location of things were spread out or doubled in the code base. This package aggregates them in one place.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Review by MiguelPires — APPROVED** [review]
LGTM

**Review by pedronis — COMMENTED** [review]
some small remarks. I can address them in a bit if nobody gets to them

**Review by mardy — APPROVED** [review]
Thanks Michael for the updates! I noticed something in the tests, but it's a nitpick.

File: gadget/device/encrypt_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		filepath.Join(d, boot.InstallHostFDESaveDir), []byte("foo"))
+	c.Assert(err, IsNil)
+	// both markers were written
+	c.Check(filepath.Join(d, boot.InstallHostFDEDataDir, "marker"), testutil.FileEquals, "foo")
+	c.Check(filepath.Join(d, boot.InstallHostFDESaveDir, "marker"), testutil.FileEquals, "foo")
+	// and can be read with device.ReadEncryptionMarkers
+	m1, m2, err := device.ReadEncryptionMarkers(filepath.Join(d, boot.InstallHostFDEDataDir), filepath.Join(d, boot.InstallHostFDESaveDir))
+	c.Assert(err, IsNil)
+	c.Check(m1, DeepEquals, []byte("foo"))
+	c.Check(m2, DeepEquals, []byte("foo"))
```
[the remark this answers is not in the document]
Comment (mvo5) [reply gadget/device/encrypt_test.go:64]: Added

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**